### PR TITLE
fix(router): enforce team scope and fail closed on missing credentials

### PR DESCRIPTION
**Pull request by yujonglee-berri**
## TLDR

Problem this investigates:

- Exact deployment IDs bypass team ownership checks in multiple routing paths
- Missing and empty named credentials can discard usable inline auth and deployment metadata
- Public files and batch paths do not consistently pass authenticated team scope into credential lookup

The narrow scoped exact-ID lookup is fixed, and the test-only contract matrix documents 16 unresolved gaps as strict xfails. Production fixes remain

## User Flow

Before: a developer using team-scoped credential lookup can receive another team's exact deployment credentials

1. They configure an exact deployment ID owned by team B
2. They resolve that ID with team A's authenticated team ID
3. The lookup returns team B's API key

After the narrow fix: the same scoped Router lookup respects authenticated team ownership

1. They configure the same exact deployment ID owned by team B
2. They resolve that ID with team A's authenticated team ID
3. The lookup returns no credentials for that deployment

The public files and batch call sites still need to propagate that scope, and rejected exact IDs must not fall through to wildcard credentials

## Pre-Submission checklist

- [x] I have added meaningful tests
- [x] The handful of test files covering my change pass locally
- [ ] My PR passes all required CI/CD checks
- [ ] I have received a Greptile **Confidence Score of at least 4/5** before requesting a maintainer review

## Delays in PR merge?

If you're seeing a delay in your PR being merged, ping the LiteLLM Team on [Slack (#pr-review)](https://join.slack.com/t/litellmossslack/shared_invite/zt-3o7nkuyfr-p_kbNJj8taRfXGgQI1~YyA)

## Screenshots / Proof of Fix

The same Router credential-resolution script was run at both commits with team-owned and missing-credential deployments

#### Cross-team exact ID

1. Resolve team B's exact deployment ID with `team_id="team-a"`
2. Observe `{'api_key': 'team-b-key', 'model': 'openai/gpt-5.2', 'custom_llm_provider': 'openai'}`

#### Missing named credential

1. Resolve a deployment referencing a deleted named credential
2. Observe `{'model': 'azure/text-embedding-3-small', 'custom_llm_provider': 'azure'}`

#### Unscoped exact-ID compatibility

1. Resolve team B's exact deployment ID without `team_id`
2. Observe team B's configured credentials

#### Cross-team exact ID

1. Resolve team B's exact deployment ID with `team_id="team-a"`
2. Observe `None`

#### Missing named credential

1. Resolve a deployment referencing a deleted named credential
2. Observe `None`

#### Unscoped exact-ID compatibility

1. Resolve team B's exact deployment ID without `team_id`
2. Observe team B's configured credentials unchanged

## Current test status

- 583 tests pass across the three changed suites
- 16 strict xfails document unresolved regression contracts
- 14 unrelated tests skip under the local environment

The xfails cover inline credential fallback, ambient auth, S3 metadata, endpoint team propagation, exact-ID terminal denial, team-public-name precedence, and wildcard fallback

## Type

- Bug Fix
- Test

### High

- Authenticated files and batch routes can resolve team-owned exact IDs without passing the caller's team scope
- Authenticated teamless callers still reach the legacy unscoped exact-ID path

### Medium

- Empty saved credentials are indistinguishable from missing credentials
- Rejected exact IDs can be reinterpreted through a shared wildcard
- Team public names can lose to colliding shared model names
- A more-specific foreign wildcard can shadow a broader shared wildcard
- Normal Router exact-ID and wildcard routing has adjacent team-scope gaps

### Low

- Credential expiration validation is outside this PR's scope

## Final Attestation

- [ ] The tests check the right things, including the edge cases, and regressions in the respective real-world customer use-cases are not possible after this PR

### CODE REVIEW

File: litellm/router.py [hunk 1]
```diff
@@ -9574,6 +9574,8 @@ def get_deployment_credentials_with_provider(
         """
         # Try to get deployment by model_id first
         deployment = self.get_deployment(model_id=model_id)
+        if deployment is not None and team_id is not None and not self._deployment_usable_by_team(deployment, team_id):
+            deployment = None
```
[the remark this answers is not in the document]
Comment (yujonglee-berri) [reply litellm/router.py]: Refactored this with a named Final candidate instead of the walrus expression, preserving behavior while avoiding the new reassignment

**Comment by yujonglee-berri** [comment]
@greptile-apps Review
